libs/designsystem/modal/src/modal-navigation.service.ts

```diff
@@ -340,4 +340,15 @@ export class ModalNavigationService {
     }
     return parentRoute;
   }
+
+  handleBackButton(modal: HTMLIonModalElement) {
+    const popStateEvent$ = fromEvent(window, 'popstate');
+    const popStateSubscription = popStateEvent$.subscribe(() => {
+      modal.dismiss();
+    });
+
+    modal.onDidDismiss().then(() => {
+      popStateSubscription.unsubscribe();
+    });
```

Comment:
Consider simplifying 😊 

`takeUntil` ensures the observables completes when the modal closes and therefore we don't need to unsubscribe from the stream 🎉

See this detailed explanation: https://benlesh.medium.com/rxjs-dont-unsubscribe-6753ed4fda87 
And docs for `takeUntil`: https://rxjs.dev/api/index/function/takeUntil
```suggestion
    // convert Promise to Observable:
    const modalClose$ = from(modal, modal.onDidDismiss());
    // OR convert Event to Observable:
    // const modalClose$ = fromEvent(modal, 'ionModalDidDismiss');
    popStateEvent$.takeUntil(modalClose$).subscribe(() => {
      modal.dismiss();
    });
```